app/screens/search/search.js

```diff
@@ -467,7 +468,9 @@ export default class Search extends PureComponent {
             },
         });
 
-        actions.searchPosts(currentTeamId, terms.trim(), isOrSearch, true);
+        // timezone offset in seconds
+        const timeZoneOffset = getDeviceUtcOffset() * 60;
```

Comment:
If the timezone feature is enabled you should use that one and then fallback to the device